lib/3scale/backend/listener.rb

```diff
@@ -457,11 +451,12 @@ def valid_usage_params?
         params[:usage].nil? || params[:usage].is_a?(Hash)
       end
 
-      def check_params_value_encoding!(input_params, params_to_validate)
-        params_to_validate.each do |p|
-          param_value = input_params[p]
-          if !param_value.nil? && !param_value.valid_encoding?
-            halt 400, ThreeScale::Backend::NotValidData.new.to_xml
+      def check_params_encoding!(input_params)
```

Comment:
So, you're now checking all parameters, not only the list predefined in `REPORT_EXPECTED_PARAMS` ?